### PR TITLE
Various NetCore improvements

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
- Added NetStandard to Remote.TestKits projects
- Updated F# dependencies
- Use a global version of Test.SDK
- Added NetCore build for Performance projects

### CODE REVIEW

File: build.fsx [hunk 1]
```diff
@@ -110,19 +110,11 @@ Target "RunTests" (fun _ ->
...
-        | _ -> !! "./**/core/**/*.Tests.csproj"
-                  ++ "./**/contrib/**/*.Tests.csproj"
-                  -- "./**/serializers/**/*Wire*.csproj"
-                  -- "./**/Akka.MultiNodeTestRunner.Shared.Tests.csproj"
-                  -- "./**/Akka.API.Tests.csproj"
+    let projects = !! "./**/core/**/*.Tests.csproj"
+                   ++ "./**/contrib/**/*.Tests.csproj"
+                   -- "./**/serializers/**/*Wire*.csproj"
+                   -- "./**/Akka.MultiNodeTestRunner.Shared.Tests.csproj"
+                   -- "./**/Akka.API.Tests.csproj"
```
Comment (alexvaluyskiy) [inline build.fsx]: `Akka.API.Tests` should be disabled here too

File: src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingSpec.cs [hunk 2]
```diff
@@ -300,7 +300,7 @@ private void CreateCoordinator()
 
             foreach (var typeName in typeNames)
             {
-                var rebalanceEnabled = string.Equals(typeName, "rebalancing", StringComparison.InvariantCultureIgnoreCase);
+                var rebalanceEnabled = string.Equals(typeName, "rebalancing", StringComparison.OrdinalIgnoreCase);
```
Comment (alexvaluyskiy) [inline src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingSpec.cs]: `InvariantCultureIgnoreCase` does not work on NetCore

File: src/core/Akka.FSharp/Akka.FSharp.fsproj [hunk 3]
```diff
@@ -20,24 +21,9 @@
...
-    </Reference>
-    <Reference Include="FsPickler">
-      <HintPath>..\..\packages\FsPickler.3.2.0\lib\net45\FsPickler.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
-    <Reference Include="mscorlib" />
-    <PackageReference Include="FSharp.Core" Version="4.1.0" />
-    <PackageReference Include="FSharp.NET.Sdk" Version="1.0.1" PrivateAssets="All" />
-    <PackageReference Include="FsPickler" Version="1.2.21" />
+    <PackageReference Include="FSharp.Core" Version="4.2.1" />
```
Comment (alexvaluyskiy) [inline src/core/Akka.FSharp/Akka.FSharp.fsproj]: Updated `FSharp.Core` dep

File: src/core/Akka.Remote.TestKit/CommandLine.cs [hunk 4]
```diff
@@ -26,26 +26,26 @@ namespace Akka.Remote.TestKit
     /// </summary>
     public class CommandLine
     {
-        private static readonly Lazy<StringDictionary> Values = new Lazy<StringDictionary>(() =>
+        private static readonly Lazy<Dictionary<string, string>> Values = new Lazy<Dictionary<string, string>>(() =>
```
Comment (alexvaluyskiy) [inline src/core/Akka.Remote.TestKit/CommandLine.cs]: Moved from non-generic collection

File: src/core/Akka.Remote.TestKit/MultiNodeSpec.cs [hunk 5]
```diff
@@ -377,6 +377,7 @@ public static Config BaseConfig
 
         private static string GetCallerName()
         {
+#if !CORECLR
```
Comment (alexvaluyskiy) [inline src/core/Akka.Remote.TestKit/MultiNodeSpec.cs]: It is a hack. Could not be fixed on NetCore 1.x

File: src/core/Akka.Tests/IO/TcpConnectionSpec.cs [hunk 6]
```diff
@@ -25,7 +25,7 @@
 
 namespace Akka.Tests.IO
 {
-    class TcpConnectionSpec : AkkaSpec
+    public class TcpConnectionSpec : AkkaSpec
```
Comment (alexvaluyskiy) [inline src/core/Akka.Tests/IO/TcpConnectionSpec.cs]: All test classes should be public

File: src/common.props [hunk 7]
```diff
@@ -11,6 +11,7 @@
 
   <PropertyGroup>
     <XunitVersion>2.3.0-beta2-*</XunitVersion>
+    <TestSDKVersion>15.0.0</TestSDKVersion>
```
Comment (alexvaluyskiy) [inline src/common.props]: Use a constant, in order to easy update the version

**Comment by Aaronontheweb** [comment]
@alexvaluyskiy needs a rebase

**Pull request closed without merging**